packages/flutter_driver/lib/src/driver.dart

```diff
@@ -545,12 +549,38 @@ class FlutterDriver {
   ///
   /// [streams] limits the recorded timeline event streams to only the ones
   /// listed. By default, all streams are recorded.
-  Future<Timeline> traceAction(Future<dynamic> action(), { List<TimelineStream> streams: _defaultStreams }) async {
+  ///
+  /// If [retainPriorEvents] is true, retains events recorded prior to calling
+  /// [action]. Otherwise, prior events are cleared before calling [action]. By
+  /// default, prior events are cleared.
+  Future<Timeline> traceAction(
+    Future<dynamic> action(), {
+    List<TimelineStream> streams: _defaultStreams,
+    bool retainPriorEvents: false,
+  }) async {
+    if (!retainPriorEvents) {
+      await clearTimeline();
+    }
     await startTracing(streams: streams);
     await action();
     return stopTracingAndDownloadTimeline();
   }
 
+  /// Clears all timeline events recorded up until now.
+  Future<Null> clearTimeline({ Duration timeout: _kShortTimeout }) async {
```

Comment:
`@visibleForTesting`?

Reply:
No, I think it's useful to be in the public API.